src/global_scheduler/global_scheduler_algorithm.cc

```diff
@@ -24,6 +24,13 @@ void GlobalSchedulerPolicyState_free(GlobalSchedulerPolicyState *policy_state) {
  */
 bool constraints_satisfied_hard(const LocalScheduler *scheduler,
                                 const TaskSpec *spec) {
+  if (scheduler->info.static_resources.at("CPU") == 0) {
```

Comment:
Maybe not in this PR, but It would be great if we could store the error for when there are no feasible nodes as part of the return value for the task. How doable is that?

Reply:
Hm, what do you mean by "as part of the return value for the task? E.g., store an exception 
object in the object store so that `ray.get` on the return value raises an exception? I think something like this would be useful. Right now we just print an error to STDERR at https://github.com/ray-project/ray/blob/0fcceef772a3126e027b0817988022ebda1a506e/src/global_scheduler/global_scheduler_algorithm.cc#L139-L140.

It's possible for the lack of feasible nodes to be transient (e.g., if more nodes are being added to the cluster or some were just slow to start up).

Reply:
Hmm, I didn't think about more nodes getting added to the cluster. Yeah, I was thinking about the possibility of storing the exception, but I agree, I think the printed error is good then.